geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -5127,9 +5127,15 @@ private void basicSetPdxSerializer(PdxSerializer serializer) {
     }
   }
 
+  @Override
+  public void setReadSerializedForCurrentThread(boolean value) {
+    PdxInstanceImpl.setPdxReadSerialized(value);
+    DefaultQuery.setPdxReadSerialized(value);
+  }
+
   // test hook
   @Override
-  public void setReadSerialized(boolean value) {
+  public void setReadSerializedForTest(boolean value) {
```

Comment:
Can we deprecate setReadSerializedForTest and tell people not to use it? I can't really see a valid use case for changing this value on the fly. I do see some existing yucky tests that are trying to test 20 things without closing the cache that seem to be using this method.

Reply:
I'll add a deprecation annotation